src/graphics/material.cpp

```diff
@@ -60,13 +60,28 @@ Material::Material(const XMLNode *node, bool deprecated)
     m_installed = false;
 
     m_sampler_path[0] = "unicolor_white";
-    node->get("name",      &m_texname);
-    if (m_texname=="")
+    node->get("name",      &m_matname);
+    if (m_matname=="")
     {
-        throw std::runtime_error("[Material] No texture name specified "
+        throw std::runtime_error("[Material] No material name specified "
                                  "in file\n");
     }
 
+    node->get("tex-layer-0",      &m_texname);
+    if (m_texname=="")
+    {
+        // We fallback to using the name as the slot 0 texture
+        node->get("name",      &m_texname);
+        if (m_texname=="")
+        {
+            throw std::runtime_error("[Material] No material name specified "
+                                     "in file\n");
```

Comment:
This error-checking doesn't make much sense. You already try to get the "name" node value for m_matname, and throw an error if there is none, so this part of the code will never be reached.

There is also no need to retrieve again the name node.

You could replace all this by a simple : 
m_texname = m_matname;

Reply:
O="hd_palmTreeLeaf_alAlbedo.png"/>